### PR TITLE
UPDATE gems

### PR DESCRIPTION
**Pull request by cdmo**
* ast 2.4.0 → 2.4.1
[changelog](https://github.com/whitequark/ast/blob/master/CHANGELOG.md)

* coderay 1.1.2 → 1.1.3
[changelog](https://github.com/rubychan/coderay/blob/master/Changes-pre-1.0.textile)

* concurrent-ruby 1.1.6 → 1.1.7
[changelog](https://github.com/ruby-concurrency/concurrent-ruby/blob/master/CHANGELOG.md#release-v117-6-august-2020)

* diff-lcs 1.3 → 1.4.4
[changelog](https://github.com/halostatue/diff-lcs/blob/master/History.md#144--2020-07-01)

* dry-initializer 3.0.3 → 3.0.4
[changelog](https://github.com/dry-rb/dry-initializer/blob/master/changelog.yml)

* dry-logic 1.0.7 → 1.0.8
[changelog](https://github.com/dry-rb/dry-logic/blob/master/changelog.yml)

* dry-schema 1.5.2 → 1.5.5
[changelog](https://github.com/dry-rb/dry-schema/blob/master/changelog.yml)

* dry-validation 1.5.3 → 1.5.6
[changelog](https://github.com/dry-rb/dry-validation/blob/master/changelog.yml)

* ffi 1.13.0 → 1.13.1
[changelog](https://github.com/ffi/ffi/blob/master/CHANGELOG.md#1131--2020-06-09)

* nokogiri 1.10.9 → 1.10.10
[changelog](https://github.com/sparklemotion/nokogiri/blob/master/CHANGELOG.md#11010--2020-07-06)

* parallel 1.19.1 → 1.19.2

* parser 2.7.1.2 → 2.7.2.0
[changelog](https://github.com/whitequark/parser/blob/v2.7.2.0/CHANGELOG.md)

* public_suffix 4.0.5 → 4.0.6
[changelog](https://github.com/weppos/publicsuffix-ruby/blob/v4.0.6/CHANGELOG.md#406)

* rspec-core 3.9.2 → 3.9.3
[changelog](https://github.com/rspec/rspec-core/blob/main/Changelog.md)

* rubocop 0.84.0 → 0.93.1
[changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md#0931-2020-10-12)

* rubocop-ast 0.0.3 → 0.8.0
[changelog](https://github.com/rubocop-hq/rubocop-ast/blob/master/CHANGELOG.md#080-2020-10-12)

* traject 3.3.0 → 3.4.0
[changelog](https://github.com/traject/traject/blob/master/CHANGES.md#340)

### CODE REVIEW

**Comment by cdmo** [comment]
Putting this update on hold pending finding a scenario or two or three that will trigger "CD" or "Vinyl disc" or both in `MarcMediaTypeProcessor::resolve_300`

**Comment by cdmo** [comment]
ok this is ready to go for real this time

File: lib/traject/macros/custom.rb [hunk 1]
```diff
@@ -99,7 +101,7 @@ def extract_link_data(link_type: 'full')
             link_data << collect_subfield_values(field: field, code: 'u')
           end
           link_data.flatten.compact.each do |link|
-            url_match = regex_split link, %r{https*://([\w*|\.*]*)}
+            url_match = regex_split link, %r{https*://([\w*|.*]*)}
```
Comment (banukutlu) [inline lib/traject/macros/custom.rb:104]: no need for `\` ?

Comment (cdmo) [reply lib/traject/macros/custom.rb:104]: This was fixed automatically by: https://docs.rubocop.org/rubocop/cops_style.html#styleredundantregexpescape

RubyMine warns too:

Screen Shot 2020-10-26 at 3 48 42 PM

I was unable to find ruby lang docs supporting this issue. I did test this in rubular though and it seems to check out

Comment (cdmo) [reply lib/traject/macros/custom.rb:104]: Had no idea escaping. Kinda wild:

```
irb(main):012:0> a = /a[.]c/
=> /a[.]c/
irb(main):013:0> b = /a.c/
=> /a.c/
irb(main):014:0> 'abc'.match? a
=> false
irb(main):015:0> 'abc'.match? b
=> true
irb(main):016:0> 'a.c'.match? a
=> true
irb(main):017:0> 'a.c'.match? b
=> true
irb(main):018:0> c = /a[\.]c/
=> /a[\.]c/
irb(main):019:0> 'abc'.match? c
=> false
irb(main):020:0> 'a.c'.match? c
=> true
```

Also found this, see under ["Metacharacters Inside Character Classes"](https://www.regular-expressions.info/charclass.html)

Comment (cdmo) [reply lib/traject/macros/custom.rb:104]: With that knowledge, then there are other flaws with this regex. For example, the `*` is taken as a literal, not a metacharacter of "zero or more of the previous character"

Comment (banukutlu) [reply lib/traject/macros/custom.rb:104]: oh yes, `*` is a literal in character class. so the below seems to still work? tried with rubular:

`%r{https*://([\w|.]*)}`

File: lib/traject/macros/custom.rb [hunk 2]
```diff
@@ -99,7 +101,7 @@ def extract_link_data(link_type: 'full')
             link_data << collect_subfield_values(field: field, code: 'u')
           end
           link_data.flatten.compact.each do |link|
-            url_match = regex_split link, %r{https*://([\w*|\.*]*)}
+            url_match = regex_split link, %r{https*://([\w.]*)}
```
Comment (cdmo) [inline lib/traject/macros/custom.rb:104]: k this is better

File: lib/traject/macros/custom.rb (hunk 1 above)
Comment (cdmo) [reply lib/traject/macros/custom.rb]: the pipe is a literal too. that area inside the square brackets treats all characters as "OR"

so it reads like "[any word character OR a pipe OR a dot]" zero or more times following the "//" and capture all that

File: lib/traject/macros/custom.rb (hunk 2 above)
Comment (banukutlu) [reply lib/traject/macros/custom.rb:104]: yup 👍

File: lib/traject/macros/custom.rb (hunk 1 above)
Comment (banukutlu) [reply lib/traject/macros/custom.rb]: yes that makes sense 👍

**Review by banukutlu — APPROVED** [review]
👍

**Pull request merged**